File: jobportal-backend/src/main/java/com/jobportal/service/JobApplicationService.java
package com.jobportal.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.jobportal.model.JobApplication;
import com.jobportal.repository.JobApplicationRepository;

@Service
public class JobApplicationService {

    @Autowired
    private JobApplicationRepository jobApplicationRepository;

    // Apply for a job
    public String applyJob(JobApplication application) {
        jobApplicationRepository.save(application);
        return "Job Applied Successfully!";
    }
}

File: src/services/JobApplicationService.js
import axiosInstance from "./axios";

const JobApplicationService = {
  apply: async (data) => {
    console.log("📤 Applying job with data:", data);
    const response = await axiosInstance.post("/application/apply", data);
    return response.data;
  },
};

export default JobApplicationService;
